cmd/submit_test.go

```diff
@@ -112,6 +123,9 @@ func TestSubmit(t *testing.T) {
 	err = apiCfg.Write()
 	assert.NoError(t, err)
 
+	// write some mock input for interactive commands
```

Comment:
I don't understand why the mock input needs to be written to the file. Would you elaborate?

Reply:
Sure - perhaps there is a better solution than what I came up with.

The challenge for interactive prompt testing was that `cmd\submit.go` uses `comms.Question` to ask the user for input, and stdin is used as the input source. As I understand, `go test` overrides stdin to point to dev/null (or something that is empty), so the only response given to the prompts from stdin during the tests was `EOF`.

All that the prompt needs to get input is a file interface (`Open` and `Read` methods), and it expects to read the response to each prompt one line at a time, so writing the prompts delimited by newline to a temp file and reading from that, one line at a time, provided the necessary interface and can be easily cleaned up afterwards.

I'll see if some more comments could clarify that intent.

Reply:
Ah, and reading this again I see another thing -- I wasn't sure if it was better to have the test tmpFile setup and teardown handled in `cmd_test.go` or in the test file itself. On one hand, it would be nice to not have to manually set that up each time. On the other hand, input/output redirection is already happening in the test file, so perhaps it makes sense to put there with other similar tasks.

Reply:
I *think* you can override Out so that it goes to something that you capture instead of going to /dev/null, which would let you avoid the mock, but still get the same benefit.

With respect to the test setup and teardown, I want each test to do it for clarity, and so that we can customize. (I might be missing something here, though—if what I'm saying makes no sense, feel free to push back!)

Reply:
Sorry. I mean override `In`, not override `Out`.

So in your test, you would set `In` to `strings.NewReader("your\nfake\ninput")`. I think.

Reply:
Ok, looking through all of this one more time (so sorry about the noise)... I think that it makes sense to provide the `MockInput` on the `CommandTest` value, but instead of writing it to a file, it would set `In` to be a reader that contains that mock input.

I think we might also want to rename `MockInput` to `FakeInteractiveResponse` for clarity.